src/vnm_mad/remotes/lib/vm.rb

```diff
@@ -121,6 +122,27 @@ def nic_build_hash(nic_element, nic)
                 end
             end
 
+            # Method to build the list of NIC/NIC_ALIAS
+            #   @param xpath_filter [String] XML NIC/NIC_ALIAS document
+            def nics_build(xpath_filter)
+                nics = VNMNetwork::Nics.new(hypervisor)
+
+                @vm_root.elements.each(xpath_filter) do |nic_element|
+                    nic = nics.new_nic
+
+                    nic_build_hash(nic_element, nic)
+
+                    if !VNMMAD.pre_action?
+                        nic.get_info(self)
```

Comment:
Do these functions work/make sense for NIC_ALIAS? (when calling it as nics_build('TEMPLATE/NIC_ALIAS')